Lib/multiprocessing/resource_tracker.py

```diff
@@ -81,6 +97,9 @@ def ensure_running(self):
         This can be run from any process.  Usually a child process will use
         the resource created by its parent.'''
         with self._lock:
+            if self._lock._recursion_count() > 1:
+                # The code below is certainly not reentrant-safe, so bail out
+                return self._reentrant_call_error()
```

Comment:
I would prefer:

```suggestion
                raise self._reentrant_call_error()
```

or:

```suggestion
                self._reentrant_call_error()
```

Since the ``return`` here is a lie: the function never returns.